File: ruby/test/fizzbuzz_test.rb
require_relative "../fizzbuzz.rb"
require "test/unit"

class TestFizzBuzz < Test::Unit::TestCase

    def test_should_print_fizzbuzz_when_divisible_by_three_and_five
        assert_equal("fizzbuzz", Fizzbuzz.generate(15))
    end

    def test_should_print_fizz_when_divisible_by_only_three
        assert_equal("fizz", Fizzbuzz.generate(9))
    end

    def test_should_print_buzz_when_divisible_by_only_five
        assert_equal("buzz", Fizzbuzz.generate(10))
    end

    def test_should_print_number_as_string_when_no_divisible_by_three_or_five
        assert_equal("4", Fizzbuzz.generate(4))
    end
    
end

File: php/tests/FizzBuzzTest.php
<?php

use \PHPUnit\Framework\TestCase;

/**
 * Class FizzBuzzTest
 *
 * Test class for fizzbuzz generator
 */
class FizzBuzzTest extends TestCase
{

    /**
     * Test should return fizzbuzz when number is divisible by 3 and 5
     *
     * @return void
     */
    public function testShouldReturnFizzBuzzWhenNumberIsDivisibleByThreeAndFive()
    {
        $this->assertEquals("FizzBuzz", FizzBuzz::generate(15), 'Should return FizzBuzz');
    }

    /**
     * Test should return fizzbuzz when number is divisible by 3
     *
     * @return void
     */
    public function testShouldReturnFizzWhenNumberIsDivisibleByThree()
    {
        $this->assertEquals("Fizz", FizzBuzz::generate(9), 'Should return Fizz');
    }

    /**
     * Test should return buzz when number is divisible by 5
     *
     * @return void
     */
    public function testShouldReturnBuzzWhenNumberIsDivisibleByFive()
    {
        $this->assertEquals("Buzz", FizzBuzz::generate(10), 'Should return Buzz');
    }

    /**
     * Test should return number when number is not divisible by 3 and 5
     *
     * @return void
     */
    public function testShouldReturnNumberWhenNumberIsNotDivisibleByThreeAndFive()
    {
        $this->assertEquals("11", FizzBuzz::generate(11), 'Should return 11');
    }
}
